packages/flutter_test/lib/src/_goldens_io.dart

```diff
@@ -125,26 +125,31 @@ class LocalComparisonOutput {
   /// Writes out diffs from the [ComparisonResult] of a golden file test.
   ///
   /// Will throw an error if a null result is provided.
-  void generateFailureOutput(
+  Future<void> generateFailureOutput(
     ComparisonResult result,
     Uri golden,
     Uri basedir, {
     String key = '',
-  }) {
+  }) async {
     String additionalFeedback = '';
     if (result.diffs != null) {
       additionalFeedback = '\nFailure feedback can be found at '
         '${path.join(basedir.path, 'failures')}';
-      final Map<String, Image> diffs = result.diffs.cast<String, Image>();
-      diffs.forEach((String name, Image image) {
+      final Map<String, ByteData> diffs =
+          result.diffs.cast<String, ByteData>();
+      for (final MapEntry<String, ByteData> entry in diffs.entries) {
         final File output = getFailureFile(
-          key.isEmpty ? name : name + '_' + key,
+          key.isEmpty ? entry.key : entry.key + '_' + key,
           golden,
           basedir,
         );
         output.parent.createSync(recursive: true);
-        output.writeAsBytesSync(encodePng(image));
-      });
+        final Codec codec = await instantiateImageCodec(
```

Comment:
This code is wrong - tests are now failing with "cannot instantiate image codec" which makes sense because these are just the RGBA bytes, not the actual image bytes.

So the question becomes, how to encode rgba bytes to png bytes?

Reply:
It seems like `Image.toByteData(format: ImageByteFormat.png)` is essentially what I want but the problem is I can't fabricate an `Image` object from raw rbga.

Do we need to add that to the Image class from `dart:ui`?

Reply:
Yes, I do think that is one the things we are missing here. In a previous implementation, we compared image bytes, but it wasn't an accurate comparison. We introduced the Image package so we could do pixel comparison and create diff images.

Reply:
I pinged on the original issue about encoding to png and replaced this with a todo.